src/torchmetrics/utilities/plot.py

```diff
@@ -0,0 +1,264 @@
+            )
+    else:
+        val = torch.stack(val, 0)
+        multi_series = val.ndim != 1
+        val = val.T if multi_series else val.unsqueeze(0)
+        for i, v in enumerate(val):
+            ax.plot(
+                v.detach().cpu(),
+                marker="o",
+                markersize=10,
+                linestyle="-",
+                label=(f"{legend_name} {i}" if legend_name else f"{i}") if multi_series else None,
+            )
+        ax.get_xaxis().set_visible(True)
+        ax.set_xlabel("Step")
+        ax.set_xticks(torch.arange(val.shape[1]))
+
+    handles, labels = ax.get_legend_handles_labels()
+    if handles and labels:
+        ax.legend(handles, labels, loc="upper center", bbox_to_anchor=(0.5, 1.15), ncol=3, fancybox=True, shadow=True)
+
+    if lower_bound is not None and upper_bound is not None:
+        factor = 0.1 * (upper_bound - lower_bound)
+    else:
+        ylim = ax.get_ylim()
+        factor = 0.1 * (ylim[1] - ylim[0])
+
+    ax.set_ylim(
+        bottom=lower_bound - factor if lower_bound is not None else None,
+        top=upper_bound + factor if upper_bound is not None else None,
+    )
+
+    ax.grid(True)
+    ax.set_ylabel(name if name is not None else None)
+
+    xlim = ax.get_xlim()
+    factor = 0.1 * (xlim[1] - xlim[0])
+
+    ax.hlines(
+        [lower_bound if lower_bound is not None else None, upper_bound if upper_bound is not None else None],
+        xlim[0],
+        xlim[1],
+        linestyles="dashed",
+        colors="k",
+    )
+    if higher_is_better is not None:
+        if lower_bound is not None and not higher_is_better:
+            ax.set_xlim(xlim[0] - factor, xlim[1])
+            ax.text(
+                xlim[0], lower_bound, s="Optimal \n value", horizontalalignment="center", verticalalignment="center"
+            )
+        if upper_bound is not None and higher_is_better:
+            ax.set_xlim(xlim[0] - factor, xlim[1])
+            ax.text(
+                xlim[0], upper_bound, s="Optimal \n value", horizontalalignment="center", verticalalignment="center"
+            )
+    return fig, ax
+
+
+def _get_col_row_split(n: int) -> Tuple[int, int]:
+    """Split n curves into rows x cols figures."""
+    nsq = sqrt(n)
+    if nsq * nsq == n:
+        return nsq, nsq
+    elif floor(nsq) * ceil(nsq) > n:
+        return floor(nsq), ceil(nsq)
+    else:
+        return ceil(nsq), ceil(nsq)
+
+
+def trim_axs(axs: Union[matplotlib.axes.Axes, np.ndarray], N: int) -> np.ndarray:
+    """Reduce *axs* to *N* Axes.
+
+    All further Axes are removed from the figure.
+    """
+    if isinstance(axs, matplotlib.axes.Axes):
+        return axs
+    else:
+        axs = axs.flat
+        for ax in axs[N:]:
+            ax.remove()
+        return axs[:N]
+
+
+def plot_confusion_matrix(
+    confmat: Tensor,
+    add_text: bool = True,
+    labels: Optional[List[str]] = None,
+    cmap: Optional[matplotlib.colors.Colormap] = None,
```

Comment:
also add option for axes? Should be consistent :)

Reply:
will update